app/controllers/tags_controller.rb

```diff
@@ -49,6 +49,11 @@ def destroy
         @file.tags.delete(@tag)
     end
 
+    # If tag has no file relation, remove
+    if !@tag.bruse_files[0]
```

Comment:
`!@tag.bruse_files[0]` could be replaced with the [`empty?`-method](http://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-empty-3F) Rails provides for associative collections.

Like this:

```ruby
if @tag.bruse_files.empty?
  ...
```